src/distributions/range.rs

```diff
@@ -454,11 +466,18 @@ macro_rules! range_float_impl {
                 // Generate a value in the range [1, 2)
                 let value1_2 = (rng.$next_u() >> $bits_to_discard)
                                .into_float_with_exponent(0);
-                // Doing multiply before addition allows some architectures to
-                // use a single instruction.
```

Comment:
Before https://github.com/rust-lang-nursery/rand/search?q=port+range&type=Issues we would do `r.low + r.range * rng.gen::<$ty>()`. I think it is nice to preserve this comment that explains why we choose a different order.

Reply:
Could you please be more specific? I'm not sure the previous comment was correct. Without `fast-math` it is not correct if the compiler uses FMA.

Reply:
Ah, there is an instruction that does a true FMA, with the benefit of extra precision, and a simple instruction that produuces exactly the same results as multiply and addition. At least on ARM.

Reply:
Thanks, I'll add a comment!